File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        first=[False] *(len(A)+1)
        second=[False] *(len(A)+1)

        ans=[0]*len(A)
        for i in range (len(A)):
            first[A[i]]=True
            second[B[i]]=True
            ans[i]=self.count(first,second)
        return ans
    def count(self,f,s):
        c=0
        for i in range(len(f)):
            if f[i]==True and s[i]==True:
                c+=1
        return c

        

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int ans[]=new int[A.length];
        Set<Integer> h=new HashSet<>();
        for(int i=0;i<A.length;i++){
            int c=0;
            h.add(A[i]);
            for(int j=0;j<=i;j++){
                if(h.contains(B[j])){
                    c++;
                }
            }
            ans[i]=c;
          
            
        }
        return ans;
    }

    // public int[] findThePrefixCommonArray(int[] A, int[] B) {
    //     int ans[]=new int[A.length];
    //     for(int i=0;i<A.length;i++){
    //         int c=0;
    //         for(int j=0;j<=i;j++){
    //             for(int k=0;k<=i;k++){
    //                 if(A[j]==B[k]){
    //                     c++;
    //                 }
    //             }
    //         }
    //         ans[i]=c;
    //     }
    //     return ans;
    // }
}